rust/numaflow-monitor/src/runtime.rs

```diff
@@ -53,12 +53,49 @@ impl From<(&Status, &str, i64)> for RuntimeErrorEntry {
         // Convert status details from bytes to a string
         let details_str = String::from_utf8_lossy(details_bytes);
 
+        // Extract metadata from gRPC Status
+        let metadata = grpc_status.metadata();
+        let mut metadata_map = std::collections::HashMap::new();
+        for key_value in metadata.iter() {
+            match key_value {
+                tonic::metadata::KeyAndValueRef::Ascii(key, value) => {
+                    metadata_map.insert(
+                        key.as_str().to_string(),
+                        value.to_str().unwrap_or("invalid_utf8").to_string(),
+                    );
+                }
+                tonic::metadata::KeyAndValueRef::Binary(key, value) => {
+                    metadata_map.insert(key.as_str().to_string(), format!("{:?}", value));
+                }
+            }
+        }
+
+        // Convert HashMap to string for storage
+        let metadata_str = if metadata_map.is_empty() {
+            String::new()
+        } else {
+            metadata_map
+                .iter()
+                .map(|(k, v)| format!("{}={}", k, v))
+                .collect::<Vec<_>>()
+                .join(", ")
+        };
+
+        // Combine details and metadata for comprehensive error information
+        let combined_details = if metadata_str.is_empty() {
+            details_str.to_string()
+        } else if details_str.is_empty() {
+            format!("metadata: {}", metadata_str)
+        } else {
+            format!("{} | metadata: {}", details_str, metadata_str)
```

Comment:
In most of the cases, since the gRPC status is constructed by us (in SDKs for panic handling) and at some places in numa. I would recommend to have metadata _str first (if it exists or constructed in future) and then details_str. Since in the UI, user may have to scroll to the bottom (lets say after the stack trace) to get metadata.

```suggestion
            format!("metadata: {} | details: {}", metadata_str, details_str)
```

Reply:
Resolved in [`dd52a80`](https://github.com/numaproj/numaflow/pull/2864/commits/dd52a8027a46be645b303e9e681a939deb29705a)